src/v/transform/transform_processor.cc

```diff
@@ -42,6 +102,8 @@ processor::processor(
   , _sinks(std::move(sinks))
   , _error_callback(std::move(cb))
   , _probe(p)
+  , _consumer_transform_pipe(1)
```

Comment:
So we will only ever have one batch in flight on each stage?

If yes I am wondering whether we could simplify of the data structures. E.g.: wouldn't have to use a full queue but could use a single future or something.

Reply:
I'm likely going to make this dynamic somehow. A single batch for the reader queue means we don't **really** get pipelining, because the reader actually returns batches of batches and we really want to always try to load from storage. Additionally we won't be able to group multiple batches on the output sink. 

However I think there is an advantage to having some exchanger data structure I'll start there until the basic memory limiting is in place.

Reply:
I'm going to leave the queue for now, but will likely need to address this as part of putting memory limits in place on the transform subsystem.